kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
                 self.deselect_node(node)
             else:
                 self.clear_selection()
```

Comment:
Yes, it should deselect the last selected item on a simple touch, without `Ctrl`.

Though you were right about the other edge case, if we remove the line as I said, and then select a range of widgets with `Shift`, and then click on one of the selected widgets, everything is deselected instead of keeping the clicked widget selected and deselecting the rest. That is wrong.

Try discarding your changes and work from there, ensuring that the reselection happens only if needed. I expect it to be a change (an `if` statement) right before that line, but of course you have to test and see what really works.